src/libraries/System.Runtime.InteropServices/tests/Ancillary.Interop/Marshalling/Base.cs

```diff
@@ -0,0 +1,129 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// Types for the COM Source Generator that define the basic vtable interactions and that we would need in the COM source generator in one form or another.
+namespace System.Runtime.InteropServices.Marshalling;
+
+/// <summary>
+/// Information about a virtual method table and the unmanaged instance pointer.
+/// </summary>
+public readonly unsafe struct VirtualMethodTableInfo
```

Comment:
Up to you: it might be nice for easy exposing and reading of commit history to have separate files for each class/struct like we would in the product. Might also be annoying if we have to do a bunch of re-naming as part of API approval. :)

Reply:
Good idea, I separated them out into their own files. F2 renaming in VS should make it pretty painless if we need to rename.